db/hybrid_logical_vector.go

```diff
@@ -180,6 +180,32 @@ func NewHybridLogicalVector() *HybridLogicalVector {
 	}
 }
 
+func (hlv *HybridLogicalVector) Equals(other *HybridLogicalVector) bool {
+	if hlv.SourceID != other.SourceID {
+		return false
+	}
+	if hlv.Version != other.Version {
+		return false
+	}
+	if len(hlv.PreviousVersions) != len(other.PreviousVersions) {
+		return false
+	}
+	for k, v := range hlv.PreviousVersions {
+		if other.PreviousVersions[k] != v {
+			return false
+		}
+	}
+	if len(hlv.MergeVersions) != len(other.MergeVersions) {
+		return false
+	}
+	for k, v := range hlv.MergeVersions {
+		if other.MergeVersions[k] != v {
+			return false
+		}
+	}
```

Comment:
fyi, there's stdlib map comparison functions so you can do:

```suggestion
	if !maps.Equal(hlv.PreviousVersions, other.PreviousVersions) {
		return false
	}
	if !maps.Equal(hlv.MergeVersions, other.MergeVersions) {
		return false
	}
```